server/src/main/java/org/elasticsearch/index/shard/IndexEventListener.java

```diff
@@ -108,6 +108,14 @@ default void afterIndexRemoved(Index index, IndexSettings indexSettings, IndexRe
 
     }
 
+    /**
+     * Called during shad create before obtaining the shard lock.
+     * @param routing the routing entry that caused the shard to be created.
+     * @param indexSettings the shards index settings
+     */
+    default void beforeShardLockDuringShardCreate(ShardRouting routing, Settings indexSettings) {
```

Comment:
can we repurpose the `beforeIndexShardCreated` method? Does not look to be seriously used anywhere

Reply:
Yes, thanks, that makes the naming much nicer, done in d8695264bfdf458a47c085198ba08449943ce523